tiny_dnn/io/caffe/layer_factory_impl.h

```diff
@@ -244,15 +244,15 @@ inline std::shared_ptr<layer> create_pooling(const caffe::LayerParameter &layer,
   }
 
   if (w_pad != 0) {
-    if (w_pad == pool_size_w - 1) {
+    if (w_pad == pool_size_w - 1 || w_pad == (pool_size_w - 1) / 2) {
```

Comment:
I believe you should replace it with `if (w_pad == (pool_size_w - 1) / 2)` (same for `h_pad`)? As far as I understand, this would exactly check for `same` padding, while the current version checks for `full` padding. cc @edgarriba @beru @bhack @wangyida